hudi-common/src/main/java/org/apache/hudi/common/lock/LockProvider.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.common.lock;
+
+import org.apache.hudi.common.config.LockConfiguration;
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.Condition;
+import java.util.concurrent.locks.Lock;
+
+/**
+ * Pluggable lock implementations using this provider class.
+ */
+public abstract class LockProvider implements Lock, AutoCloseable {
+
+  private static final Logger LOG = LogManager.getLogger(LockProvider.class);
+
+  protected LockConfiguration lockConfiguration;
+
+  protected abstract void acquireLock() throws Exception;
+
+  @Override
+  public final void lock() {
+    LOG.info("Acquiring lock");
+    try {
+      acquireLock();
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+    LOG.info("Acquired lock");
+  }
+
+  @Override
+  public final void lockInterruptibly() {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public final boolean tryLock(long time, TimeUnit unit) {
```

Comment:
we should probably support this right. even of the `lock()` , since this has a timeout param?

Reply:
Had implemented this differently earlier, changed some of the contracts, take a look again please